src/utils/log.utils.ts

```diff
@@ -0,0 +1,16 @@
+import log from 'loglevel';
+
+const environment = process.env.NODE_ENV || 'dev';
```

Comment:
**suggestion(non-blocking):** this feels rather redundant since you set this environment variable likely at build time. You can already filter in browser console by log levels when you simply use `console.log`, `console.debug`. I just feel that is library is just another dependency which increases bundle size (as small as it is). A more useful logging mechanism is a context aware one (find from where the error/log is provided). But it's just my personal opinion.

Reply:
I agree.. as silly as it may sound, I did not know about `console.debug` :D And the context aware logging seems like a nice solution. Im onto it